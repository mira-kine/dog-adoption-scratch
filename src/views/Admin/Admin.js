import React from 'react';
import DogForm from '../../components/Form/DogForm';
import { useState } from 'react';
import { createDog } from '../../services/dogs';
import Header from '../../components/Header/Header';

export default function Admin() {
  const [selectedDog, setDog] = useState({});

  const updateDogValue = (key, value) => {
    selectedDog[key] = value;
    setDog({ ...selectedDog });
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    await createDog(selectedDog);
    if (selectedDog) {
      alert("You've successfully added a dog");
    } else {
      alert('Database request failed, try again');
    }
  };

  return (
    <div>
      <Header />
      <p>
        <DogForm
          selectedDog={{ ...selectedDog }}
          handleSubmit={handleSubmit}
          updateDogValue={updateDogValue}
        />
      </p>
    </div>
  );
}
