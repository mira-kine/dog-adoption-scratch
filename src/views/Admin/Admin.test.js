import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Admin from '../Admin/Admin';

test('renders Admin page', () => {
  const { container } = render(
    <MemoryRouter>
      <Admin />
    </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
});
