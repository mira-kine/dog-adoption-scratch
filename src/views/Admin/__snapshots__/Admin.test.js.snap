// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Admin page 1`] = `
<div>
  <div>
    <div>
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Home
      </a>
    </div>
    <p>
      <div
        class="form-container"
      >
        <form>
          <label>
            Name:
            <input
              name="name"
              placeholder="name"
              type="text"
              value=""
            />
          </label>
          <label>
            Age:
            <input
              name="age"
              placeholder="age"
              type="text"
              value=""
            />
          </label>
          <label>
            Breed:
            <input
              name="breed"
              placeholder="breed"
              type="text"
              value=""
            />
          </label>
          <label>
            Image:
            <input
              name="image"
              placeholder="image"
              type="text"
              value=""
            />
          </label>
          <label>
            Bio:
            <input
              name="bio"
              placeholder="bio"
              type="text"
              value=""
            />
          </label>
          <button>
            Submit
          </button>
        </form>
      </div>
    </p>
  </div>
</div>
`;
